pystac/catalog.py

```diff
@@ -639,16 +656,27 @@ def process_catalog(
         ) -> List[Callable[[], None]]:
             setter_funcs: List[Callable[[], None]] = []
 
-            cat.resolve_links()
+            if not skip_unresolved:
+                cat.resolve_links()
 
             new_self_href = _strategy.get_href(cat, _root_href, is_root)
             new_root = os.path.dirname(new_self_href)
 
-            for item in cat.get_items():
-                setter_funcs.append(process_item(item, new_root))
-
-            for child in cat.get_children():
-                setter_funcs.extend(process_catalog(child, new_root, is_root=False))
+            for link in cat.get_links():
+                if skip_unresolved and not link.is_resolved():
+                    continue
+                elif link.rel == pystac.RelType.ITEM:
+                    link.resolve_stac_object(root=self.get_root())
+                    if isinstance(link.target, pystac.Item):
+                        setter_funcs.append(process_item(link.target, new_root))
```

Comment:
What are the implications if the link target is not a pystac.Item? Was the rel type incorrect? Right now it looks like we silently move on.

Reply:
Correct. This is a slight change in behavior -- previously, we were processing the target as if it were an Item, without any check: https://github.com/stac-utils/pystac/blob/254040480ed1fcfd14599b197d12649b15de649d/pystac/catalog.py#L432-L440

In the interest of changing as little as possible, maybe I should keep the old behavior, aka just pass the target to `process_item` without checking whether it's an item? What do you think?

Reply:
Probably a better chance that something will fail if it is not an Item. I _think_ that is preferable than silently skipping it.